src/main/java/com/mongodb/hibernate/jdbc/MongoConnection.java

```diff
@@ -0,0 +1,250 @@
+        throw new NotYetImplementedSQLException(
+                "To be implemented in scope of https://jira.mongodb.org/browse/HIBERNATE-30");
+    }
+
+    @Override
+    public int getTransactionIsolation() throws SQLException {
+        throw new NotYetImplementedSQLException(
+                "To be implemented in scope of https://jira.mongodb.org/browse/HIBERNATE-30");
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // close() and isClosed()
+
+    @Override
+    public void close() throws SQLException {
+        if (!this.closed) {
+            try {
+                this.clientSession.close();
+                this.closed = true;
+            } catch (RuntimeException e) {
+                throw new SQLException("Error closing connection", e);
+            }
+        }
+    }
+
+    @Override
+    public boolean isClosed() {
+        return this.closed;
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // Statement and PreparedStatement
+
+    @Override
+    public Statement createStatement() throws SQLException {
+        return createStatement(ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String mql) throws SQLException {
+        return prepareStatement(mql, ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
+    }
+
+    @Override
+    public Statement createStatement(int resultSetType, int resultSetConcurrency) throws SQLException {
+        throw new NotYetImplementedSQLException(
+                "To be implemented in scope of https://jira.mongodb.org/browse/HIBERNATE-16");
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String mql, int resultSetType, int resultSetConcurrency)
+            throws SQLException {
+        throw new NotYetImplementedSQLException(
+                "To be implemented in scope of https://jira.mongodb.org/browse/HIBERNATE-13");
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String mql, int autoGeneratedKeys) throws SQLException {
+        throw new UnsupportedOperationSQLException("Auto-generated key from MongoDB server side unsupported");
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String mql, String[] columnNames) throws SQLException {
+        throw new UnsupportedOperationSQLException("Auto-generated key from MongoDB server side unsupported");
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // SQL99 data types
+
+    @Override
+    public Clob createClob() throws SQLException {
+        throw new NotYetImplementedSQLException();
+    }
+
+    @Override
+    public Blob createBlob() throws SQLException {
+        throw new NotYetImplementedSQLException();
+    }
+
+    @Override
+    public NClob createNClob() throws SQLException {
+        throw new NotYetImplementedSQLException();
+    }
+
```

Comment:
I am not sure whether we need to implement the above `LOB` mapping (`GridFS` won't be supported as the product doc). If not, we could simply change to unsupported exception throwing.